cocos/core/pipeline/define.ts

```diff
@@ -63,6 +63,15 @@ export enum RenderPriority {
     DEFAULT = 0x80,
 }
 
+/**
+ * @en The pipeline render type.
+ * @zh 渲染管线类型。
+ */
+export enum PipelineType {
```

Comment:
The pipeline is designed for customization, this enum just make it static. I think the pipeline scene data is a future obstacle for customized pipeline, the different operations should be placed into the pipeline logic. If it's hard to modify currently, I suggest to at least make some APIs private:

- This enum
- pipelineType in Game
- deferred related properties in pipeline-scene-data

I think we can extend PipelineSceneData in each pipeline

Reply:
> The pipeline is designed for customization, this enum just make it static. I think the pipeline scene data is a future obstacle for customized pipeline, the different operations should be placed into the pipeline logic. If it's hard to modify currently, I suggest to at least make some APIs private:
> 
> * This enum
> * pipelineType in Game
> * deferred related properties in pipeline-scene-data
> 
> I think we can extend PipelineSceneData in each pipeline

Already delete enum PipelineType and add deferred-pipeline-scene-data.